src/components/EditorSearchBar.js

```diff
@@ -68,11 +69,16 @@ const EditorSearchBar = React.createClass({
     }
 
     if (sourceText && sourceText.get("text") &&
-      selectedSource != prevProps.selectedSource) {
+      ((selectedSource != prevProps.selectedSource) ||
+      this.context.shouldLoad)) {
```

Comment:
hmm, this doesn't need to be context, it can just be `this.shouldLoad`...

I'll look more closely tomorrow